src/fosslight_util/compare_yaml.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2022 LG Electronics Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+from fosslight_util.constant import LOGGER_NAME
+from fosslight_util.parsing_yaml import parsing_yml
+
+logger = logging.getLogger(LOGGER_NAME)
+VERSION = 'version'
+LICENSE = 'license'
+NAME = 'name'
+
+
+def compare_yaml(before_file, after_file):
+    before_oss_items, _ = parsing_yml(before_file, os.path.dirname(before_file))
+    after_oss_items, _ = parsing_yml(after_file, os.path.dirname(after_file))
+
+    before_items = get_merged_item(before_oss_items)
+    after_items = get_merged_item(after_oss_items)
+
+    new_before = []
+    for bi in before_items:
+        if bi in after_items:
+            after_items.remove(bi)
+        else:
+            new_before.append(bi)
+
+    ADD = "add"
+    DELETE = "delete"
+    CHANGE = "change"
+    PREV = "prev"
+    NOW = "now"
+
+    comp_out = {ADD: [], DELETE: [], CHANGE: []}
+
+    b_name = list(set([bi[NAME] for bi in new_before]))
+    a_name = list(set([ai[NAME] for ai in after_items]))
+
+    removed_name = [bn for bn in b_name if bn not in a_name]
+    changed_name = list(set(b_name) ^ set(removed_name))
+    added_name = list(set(a_name) ^ set(changed_name))
+
+    for bi in new_before:
+        if bi[NAME] in removed_name:
+            comp_out[DELETE].append(bi)
+        elif bi[NAME] in changed_name:
+            name_find = False
+            for ci in comp_out[CHANGE]:
+                if ci[NAME] == bi[NAME]:
+                    bi.pop(NAME)
+                    ci[PREV].append(bi)
+                    name_find = True
+                    break
+            if not name_find:
+                item_info = {NAME: bi[NAME], PREV: [], NOW: []}
+                bi.pop(NAME)
+                item_info[PREV].append(bi)
+                comp_out[CHANGE].append(item_info)
+
+    for ai in after_items:
+        if ai[NAME] in added_name:
+            comp_out[ADD].append(ai)
+        elif ai[NAME] in changed_name:
```

Comment:
How about changing the inside of `elif ai[NAME] in changed_name `like this?
AS-IS:
```
 for ci in comp_out[CHANGE]:
                if ci[NAME] == ai[NAME]:
                    ai.pop(NAME)
                    ci[NOW].append(ai)
                    break
```
TO-BE:
```
            filtered = next(filter(lambda oss_dict: oss_dict[NAME] == ai[NAME], comp_out[CHANGE]), None)
            if filtered:
                ai.pop(NAME)
                filtered[NOW].append(ai)
```